source/AI.cpp

```diff
@@ -874,6 +874,11 @@ void AI::Step(const PlayerInfo &player, Command &activeCommands)
 			// is to convert completed MOVE_TO orders into HOLD_POSITION orders.
 			UpdateOrders(*it);
 		}
+
+		// Roving takes precedence over leaving the system, but the ship will default to attacking an enemy if there is one.
+		if(isPresent && personality.IsRoving() && !isStranded && !target)
+			Rove(*it, command);
+
```

Comment:
You're splitting this if/else structure in half by putting this here. This also doesn't seem like the most appropriate place for this. Seems like it'd make more sense to me to move this up to where other personalities are checked. Lines 702 to 764 check if the ship has the swarming, harvests, surveillance, or mining personalities, then tells them to do something special based on that, so I'd imagine this would make more sense up there.

Reply:
It has to be placed after the code for navigating/jumping out of the system, or it will not take precedence, correct?

Reply:
I for one, Derpy, am extremely interested in keeping code organized. But for some reason (which I haven't figured out yet) if I _don't_ put the roving if statement there, then the test ships either don't rove and just leave the system (without a continue statement), or else they just drift in space (with a continue statement).